File: java-backend/src/main/java/com/novelai/studio/entity/WorldSetting.java
package com.novelai.studio.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 世界观设定实体
 */
@Data
@TableName(value = "world_settings", autoResultMap = true)
public class WorldSetting {

    @TableId(type = IdType.ASSIGN_UUID)
    private String id;

    /**
     * 所属书籍ID
     */
    private String bookId;

    /**
     * 分类：power_system/item/location/organization/other
     */
    private String category;

    /**
     * 设定名称
     */
    private String name;

    /**
     * 设定内容
     */
    private String content;

    /**
     * 标签列表
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> tags;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: src/types/worldSetting.ts
/**
 * 世界观设定相关类型
 */

/**
 * 设定分类
 */
export type WorldSettingCategory =
  | 'power_system'    // 力量体系
  | 'item'            // 物品道具
  | 'location'        // 地点场景
  | 'organization'    // 组织势力
  | 'other'           // 其他

/**
 * 设定分类映射
 */
export const SETTING_CATEGORY_MAP: Record<WorldSettingCategory, string> = {
  power_system: '力量体系',
  item: '物品道具',
  location: '地点场景',
  organization: '组织势力',
  other: '其他'
}

/**
 * 世界观设定
 */
export interface WorldSetting {
  id: string
  bookId: string
  category: WorldSettingCategory
  name: string
  content: string
  tags: string[]
  createdAt: string
  updatedAt: string
}

/**
 * 创建设定输入
 */
export interface CreateWorldSettingInput {
  bookId: string
  category?: WorldSettingCategory
  name: string
  content?: string
  tags?: string[]
}

/**
 * 更新设定输入
 */
export interface UpdateWorldSettingInput {
  category?: WorldSettingCategory
  name?: string
  content?: string
  tags?: string[]
}

/**
 * 设定统计
 */
export interface WorldSettingStats {
  power_system: number
  item: number
  location: number
  organization: number
  other: number
  total: number
}
